server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1329,6 +1330,11 @@ private Pair<List<Long>, Integer> searchForUserVMIdsAndCount(ListVMsCmd cmd) {
             }
         }
 
+        boolean requestingOnlyLeasedInstances = cmd.getOnlyLeasedInstances() != null && cmd.getOnlyLeasedInstances();
+        if (!VMLeaseManagerImpl.InstanceLeaseEnabled.value() && requestingOnlyLeasedInstances) {
+            throw new InvalidParameterValueException("Enable lease feature to use leased=true");
```

Comment:
Maybe reword the message to something like: Cannot list leased instances because the Instance Lease feature is disabled, please enable it to list leased instances